src/client/src/tscParseLineProtocol.c

```diff
@@ -2077,6 +2078,11 @@ static int32_t parseSmlValue(TAOS_SML_KV *pKV, const char **idx,
     val_rqoute
   } val_state;
 
+  value = malloc(bufSize);
+  if (value == NULL) {
```

Comment:
Since the init value of bufSize is TSDB_FUNC_BUF_SIZE(i.e. 512) which is not zero, ENMEM happens if the NULL pointer is returned by `malloc`.